src/axios-har-tracker.ts

```diff
@@ -1,112 +1,204 @@
-const axios = require('axios').default;
-import { writeFileSync } from 'fs';
+import { AxiosStatic } from 'axios';
 import * as cookie from 'cookie';
 
-const date = new Date();
-const startDate = date.toISOString();
-
-let generatedHar = {
+interface HarFile {
   log: {
-    version: '1.2',
+    version: string,
     creator: {
-      name: 'axios-tracker',
-      version: '1.0.0'
+      name: string,
+      version: string
     },
     pages: [],
-    entries: []
+    entries: NewEntry[];
   }
 };
 
-let newEntry = {
+interface NewEntry {
   request: {},
   response: {},
-  startedDateTime: startDate,
-  time: -1,
+  startedDateTime: string,
+  time: number,
   cache: {},
   timings: {
-    blocked: -1,
-    dns: -1,
-    ssl: -1,
-    connect: -1,
-    send: 10,
-    wait: 10,
-    receive: 10,
-    _blocked_queueing: -1
+    blocked: number,
+    dns: number,
+    ssl: number,
+    connect: number,
+    send: number,
+    wait: number,
+    receive: number,
+    _blocked_queueing: number
   }
 };
 
-async function axiosTracker(call) {
-  axios.interceptors.request.use(
-    async config => {
-      config.validateStatus = function () {
-        return true;
-      };
-      config.headers['request-startTime'] = process.hrtime();
-      const fullCookie = JSON.stringify(config.headers['Cookie']);
-      const version = config.httpVersion === undefined ? 'HTTP/1.1' : 'HTTP/' + config.httpVersion;
-
-      newEntry.request = {
-        method: config.method,
-        url: config.url,
-        httpVersion: version,
-        cookies: getCookies(fullCookie),
+export class AxiosHarTracker {
+
+  private axios: AxiosStatic;
+  private generatedHar: HarFile;
+  private newEntry: NewEntry;
+  private date = new Date().toISOString();
```

Comment:
Here you are defining property `date` and immediately assigning it's value. Value should be assigned in constructor, not here.